plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue

```diff
@@ -0,0 +1,618 @@
+          jsCodeTextarea.effect('highlight', {
+            complete: () => {
+              this.isHighlighting = false;
+            },
+          }, 1500);
+        }
+      });
+    },
+    updateCurrentSiteInfo() {
+      if (!this.hasManySiteUrls) {
+        // we make sure to disable cross domain if it has only one url or less
+        this.crossDomain = false;
+      }
+    },
+    addCustomVar() {
+      if (this.canAddMoreCustomVariables) {
+        this.customVars.push({ name: '', value: '' });
+      }
+
+      this.canAddMoreCustomVariables = !!this.maxCustomVariables
+        && this.maxCustomVariables > this.customVars.length;
+    },
+    onCustomVarNameKeydown(event: KeyboardEvent, index: number) {
+      setTimeout(() => {
+        this.customVars[index].name = (event.target as HTMLInputElement).value;
+        this.updateTrackingCode();
+      });
+    },
+    onCustomVarValueKeydown(event: KeyboardEvent, index: number) {
+      setTimeout(() => {
+        this.customVars[index].value = (event.target as HTMLInputElement).value;
+        this.updateTrackingCode();
+      });
+    },
+  },
+  computed: {
+    hasManySiteUrls() {
+      const { site } = this;
+      return this.siteUrls[site.id] && this.siteUrls[site.id].length > 1;
+    },
+    currentSiteHost() {
+      const siteUrl = this.siteUrls[this.site.id]?.[0];
+      if (!siteUrl) {
+        return '';
+      }
+
+      return getHostNameFromUrl(siteUrl);
+    },
+    currentSiteAlias() {
+      const defaultAliasUrl = `x.${this.currentSiteHost}`;
+      const alias = this.siteUrls[this.site.id]?.[1];
+      return alias || defaultAliasUrl;
+    },
+    jsTrackingIntro3a() {
+      return translate(
+        'CoreAdminHome_JSTrackingIntro3a',
+        '<a href="https://matomo.org/integrate/" rel="noreferrer noopener" target="_blank">',
+        '</a>',
+      );
+    },
+    jsTrackingIntro3b() {
+      return translate('CoreAdminHome_JSTrackingIntro3b');
+    },
+    jsTrackingIntro4a() {
+      return translate(
+        'CoreAdminHome_JSTrackingIntro4',
+        '<a href="#image-tracking-link">',
+        '</a>',
+      );
+    },
+    jsTrackingIntro5() {
+      return translate(
+        'CoreAdminHome_JSTrackingIntro5',
+        '<a rel="noreferrer noopener" target="_blank" '
+        + 'href="https://developer.matomo.org/guides/tracking-javascript-guide">',
+        '</a>',
+      );
+    },
+    mergeSubdomainsDesc() {
+      return translate(
+        'CoreAdminHome_JSTracking_MergeSubdomainsDesc',
+        'x.<span class=\'current-site-host\'></span>',
+        'y.<span class=\'current-site-host\'></span>',
+      );
+    },
+    learnMoreText() {
+      const subdomainsLink = 'https://developer.matomo.org/guides/tracking-javascript-guide'
+        + '#measuring-domains-andor-sub-domains';
+      return translate(
+        'General_LearnMore',
+        ` (<a href="${subdomainsLink}" rel="noreferrer noopener" target="_blank">`,
+        '</a>',
```

Comment:
The closing bracket got lost.
```suggestion
        '</a>)',
```